### DISCUSSION TITLE
How to filter by the last record of a field in a related table?

### DISCUSSION THREAD

**Discussion by kharikbo** [prisma/prisma#18641] [Q&A] [answered]
There are two tables in the relationship one-to-many

```
model A {
  id  Int   @id @default(autoincrement())
  B   B[]
}

model B {
  id            Int          @id @default(autoincrement())
  a             A            @relation(fields: [rId], references: [id])
  rId           Int              
  success       Boolean    
}

```

It is necessary to get an A record by `id` with the condition that the last record in the related table B will satisfy the `success = true` (without use `$queryRaw`).

How to do it? thank you in advance!

**Comment by nurul3101** [comment]
Hello @brights1de 👋 

Can you try this query?

```ts
const aRecordWithLastSuccessBRecord = await prisma.a.findUnique({
  where: { id: <a_record_id> },
  include: {
    B: {
      where: { success: true },
      orderBy: { id: 'desc' },
      take: 1,
    },
  },
});
```

This will return an A record with its related B records filtered by `success = true`, sorted by id in descending order, and limited to the first record in the sorted list that satisfies the `success = true` condition.

**Reply by nurul3101** [reply]
@brights1de If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by nurul3101** [reply]
@brights1de Do let us know in case you need any assistance.

**Reply by kharikbo** [reply]
@nurul3101 Hello!

Thanks! But I already had to change the table structure.

Actually I think that the problem is still with your approach, that "include" is selected and not "select". In "include", I can't control the fields of the related table "B" and for example other tables related to it.

Is there a way using "select" to solve my problem?

**Reply by nurul3101** [reply]
If I understand correctly, you want to replace include with select in the query, right?

**Reply by kharikbo** [reply]
@nurul3101 you are right

**Reply by nurul3101** [reply, marked as answer]
You can replace include with select and that should also work
```ts
  const aRecordWithLastSuccessBRecord = await prisma.a.findUnique({
    where: { id: <id> },
    select: {
      B: {
        where: { success: true },
        orderBy: { id: 'desc' },
        take: 1,
      },
    },
  });
```

**Reply by nurul3101** [reply]
Does it help? Let us know in case you have any other queries.
